igor-web/src/test/groovy/com/netflix/spinnaker/igor/jenkins/service/JenkinsServiceSpec.groovy

```diff
@@ -683,4 +691,62 @@ class JenkinsServiceSpec extends Specification {
         ]
     }
 
+    @Unroll
+    def "getBuildProperties retries on transient Jenkins failure"() {
+      given:
+      def jobName = "job1"
+      def buildNumber = 10
+      def artifact = new BuildArtifact()
+      artifact.displayPath = "test.properties"
+      artifact.relativePath = "test.properties"
+      artifact.fileName = "test.properties"
+      def build = new Build()
+      build.number = buildNumber
+      build.duration = 0L
+      build.artifacts = [artifact]
+      def badGatewayError = RetrofitError.httpError(
+        "http://my.jenkins.net",
+        new Response("http://my.jenkins.net", 502, "bad gateway", [], null),
+        null,
+        null
+      )
+      def propertyFile = new Response("http://my.jenkins.net", 200, "", [], new TypedString("a=b"))
+
+      when:
+      service.getBuildProperties(jobName, build.genericBuild(jobName), artifact.fileName)
+
+      then:
+      1 * client.getBuild(jobName, buildNumber) >> build
+      client.getPropertyFile(jobName, buildNumber, artifact.fileName) >> { throw badGatewayError } >> propertyFile
```

Comment:
Does this actually verify that the code retries and ends up with the correct property file? It looks like this line is just providing the mock values, since the return value in the `when` block is not getting assigned to anything/checked that the response is correct.

Reply:
Although if it was called multiple times but there was no answer for the mock, would the framework complain?

Reply:
I'm quite new, I think you are right, I'm quite new to Spock but from the docs it seems that I should not omit the cardinality (`2 *`) or it will just be a naked mock expression.

Added a check to the property file as well and verified that both actually do verify

Reply:
I always struggle with Spock, so I thought I'd ask :) Thanks for updating the tests! The behavior is much clearer now.